### PR TITLE
Refactor: lazy loading for html-wepback-plugin

### PR DESCRIPTION
**Pull request by fstylermiller**
This enables the webpack plugin to be used without a hard requirement on `html-webpack-plugin`

### CODE REVIEW

**Review by joelgallant — APPROVED** [review]
Thank you!

File: app-config-webpack/src/index.ts [hunk 1]
```diff
@@ -87,28 +86,40 @@ export default class AppConfigPlugin implements Options {
...
+        .then((module) => {
+          const HtmlWebpackPlugin = module.default;
+
+          HtmlWebpackPlugin.getHooks(compilation).alterAssetTagGroups.tapPromise(
+            'AppConfigPlugin',
+            async ({ headTags, ...html }) => {
+              // remove placeholder <script id="app-config"></script> if it exists
+              const newTags = headTags.filter(({ attributes }) => attributes.id !== 'app-config');
+
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-call
```
Comment (fstylermiller) [inline app-config-webpack/src/index.ts:99]: @joelgallant Having issues importing the `HtmlTagObject` type from a lazy loaded module. Any suggestions?

Adding this:
```js
.then((module) => {
  const HtmlWebpackPlugin = module.default;
  const { HtmlTagObject } = module;
```
Results in these errors:
```
'HtmlTagObject' is declared but its value is never read.ts(6133)
Property 'HtmlTagObject' does not exist on type '{ default: typeof HtmlWebpackPlugin; prototype: HtmlWebpackPlugin; version: number; }'.ts(2339)
'HtmlTagObject' is assigned a value but never used. Allowed unused vars must match /^_+$/u.eslint[@typescript-eslint/no-unused-vars](https://github.com/typescript-eslint/typescript-eslint/blob/v3.10.1/packages/eslint-plugin/docs/rules/no-unused-vars.md)
```

Comment (joelgallant) [reply app-config-webpack/src/index.ts:99]: `HtmlTagObject` is a type only. You can `import type { HtmlTagObject } from 'html-webpack-plugin'` up top. That won't get included in the bundle - TS erases it at runtime.

Comment (fstylermiller) [reply app-config-webpack/src/index.ts:99]: Ah perfect. Thank you

**Comment by fstylermiller** [comment]
I updated to use the logger tool you've built, as well as fixed the last of the linting errors. There are some errors I'm having locally that are preventing me from proactively fixing these issues

**Pull request merged**